react/wbooks/src/app/components/Login/index.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import { get, isArray } from 'lodash';
+import { object, objectOf } from 'prop-types';
+
+import imagePlaceholder from '../../assets/wolox-logo.png';
+import InputLabel from '../InputLabel';
+import { login } from '../../../services/User/service';
+import { Routes } from '../../../constants';
+
+import styles from './styles.module.scss';
+import { LOGIN, SIGN_UP, FIELDS, FIELDS_DATA } from './constants';
+
+class Login extends Component {
+  state = { email: '', password: '', isError: false, errorMessages: [] };
+
+  handleLogin = async event => {
+    event.preventDefault();
+    try {
+      const response = await login(this.state);
+      console.log(response);
+      this.setState({ isError: false, errorMessages: [] });
+    } catch (error) {
+      this.setState({ isError: true, errorMessages: error.data.error });
+    }
+  };
+
+  handleSignUp = () => this.props.history.push(Routes.SIGN_UP);
```

Comment:
this is available because history is injected by the `<Route` object, but what if you are in a route that doesn't  have `history` available in props?

Reply:
I have searched a lot in order to do the redirections. The only ways I could find to do it right were this and with the `withRoute`. I think with that I can do it without the history prop, right?

Reply:
Yep, you are right! thats the answer.
For now its ok!, but later we are gonna use a new library called `connected-react-router` and handle routing in redux!